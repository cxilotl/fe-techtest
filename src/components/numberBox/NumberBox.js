import React from 'react';
import PropTypes from 'prop-types';
import classNamesConcat from 'classnames';
import cssStyles from './NumberBox.module.scss';

const NumberBox = ({ numBoxLabel, onClickFn, isSelected, isHighlighted }) => {
  const name = `num-button-${numBoxLabel}`;
  let cssClasses = cssStyles.layout;
  if (isSelected) {
    cssClasses = classNamesConcat(cssClasses, cssStyles.isSelected);
  } else if (isHighlighted) {
    cssClasses = classNamesConcat(cssClasses, cssStyles.isHighlighted);
  }
  // console.log('cssClasses: ', cssClasses);
  return (
    <>
      <button type="button" name={ name } className={ cssClasses } onClick={ onClickFn } value={ numBoxLabel }>
        { numBoxLabel }
      </button>
    </>
  );
};

NumberBox.propTypes = {
  numBoxLabel: PropTypes.number,
  isSelected: PropTypes.bool,
  isHighlighted: PropTypes.bool,
  onClickFn: PropTypes.func
};

NumberBox.defaultProps = {
  isSelected: false,
  isHighlighted: false
};

export default NumberBox;