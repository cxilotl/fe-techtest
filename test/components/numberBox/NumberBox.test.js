import React from 'react';
import renderer from 'react-test-renderer';
import NumberBox from '../../../src/components/numberBox/NumberBox';

describe('NumberBox', () => {
  const labelValue = 1;

  it('Snapshot - Default button', () => {
    const component = renderer.create(<NumberBox numBoxLabel={ labelValue }/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Snapshot - Selected button', () => {
    const isSelected = true;
    const component = renderer.create(<NumberBox numBoxLabel={ labelValue } isSelected={ isSelected } />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});