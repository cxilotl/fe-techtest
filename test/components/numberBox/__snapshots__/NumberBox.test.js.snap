// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberBox Snapshot - Default button 1`] = `
<button
  className="layout"
  name="num-button-1"
  type="button"
  value={1}
>
  1
</button>
`;

exports[`NumberBox Snapshot - Selected button 1`] = `
<button
  className="layout isSelected"
  name="num-button-1"
  type="button"
  value={1}
>
  1
</button>
`;
